lib/practice_exercises.rb

```diff
@@ -1,13 +1,36 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2) where n is the size of the list.
+# Space Complexity: O(1) slice! method changes the original array. It does not change the size of the input
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  # list.each do |num|
+  #   if list.count(num) > 1
+  #     list.slice!(list.index(num))
+  #   end
+  # end
+  # return list
+  i = 0
+  list.each do |num|
+    if list[i] == list[i + 1]
+      list.slice!(list[i + 1])
+    end
+    i += 1
+  end
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n * m) n is the strings input and m is the length of the minimum string
+# Space Complexity: O(1) amount of memory used does not change as the input array changes
```

Comment:
Since the output depends on the length of the strings you could say O(1) if the strings are small or O(m) otherwise.